### PR TITLE
feat(providers): treat token expiration errors as non-retryable

### PR DESCRIPTION
**Pull request by onkarbhat**
When an AWS session token expires, kopia should treat token expiration errors as non-retryable errors. This will ensure that transfers fail fast and the error is reported back to the user sooner.

### CODE REVIEW

File: repo/blob/storage.go [hunk 1]
```diff
@@ -17,6 +17,14 @@ var ErrSetTimeUnsupported = errors.Errorf("SetTime is not supported")
 // ErrInvalidRange is returned when the requested blob offset or length is invalid.
 var ErrInvalidRange = errors.Errorf("invalid blob offset or length")
 
+// TokenExpiredErrStr is the error string returned by the provider
+// when a token has expired.
+const TokenExpiredErrStr = "The provided token has expired"
+
+// ErrTokenExpired is returned when the token used for
+// authenticating with a storage provider has expired.
+var ErrTokenExpired = errors.Errorf(TokenExpiredErrStr)
```
Comment (jkowalski) [inline repo/blob/storage.go:26]: nit: can we call this `ErrInvalidCredentials` which is possible for username/password too when the password changes. Please adjust method names everywhere.

File: repo/blob/storage.go [hunk 2]
```diff
@@ -17,6 +17,14 @@ var ErrSetTimeUnsupported = errors.Errorf("SetTime is not supported")
 // ErrInvalidRange is returned when the requested blob offset or length is invalid.
 var ErrInvalidRange = errors.Errorf("invalid blob offset or length")
 
+// TokenExpiredErrStr is the error string returned by the provider
+// when a token has expired.
+const TokenExpiredErrStr = "The provided token has expired"
```
Comment (jkowalski) [inline repo/blob/storage.go:22]: nit: does this need to be public?

File: repo/blob/storage.go (hunk 1 above)
Comment (jkowalski) [reply repo/blob/storage.go:26]: (would be also great to adjust other providers' code if we can cleanly detect credential issues there)

File: repo/blob/storage.go (hunk 2 above)
Comment (onkarbhat) [reply repo/blob/storage.go]: Yes. The const is defined in the blob package . It is used in two places: 
-  In the same blob package with the const ErrInvalidCredentials 
- In the s3 package, in the function added in this PR isInvalidCredentials()

File: repo/blob/storage.go (hunk 1 above)
Comment (onkarbhat) [reply repo/blob/storage.go]: >  can we call this ErrInvalidCredentials which is possible for username/password too when the password changes. Please adjust method names everywhere.

Done. 

> (would be also great to adjust other providers' code if we can cleanly detect credential issues there)

Can I follow up on this in future PRs ?

**Review by julio-lopez — APPROVED** [review]
We can merge and iterate. Thanks @onkarbhat

**Pull request merged**